_posts/2021-12-20-upx.adoc

```diff
@@ -0,0 +1,181 @@
+---
+layout: post
+title: 'Compressing native executables with UPX'
+date: 2021-12-20
+tags: native feature
+synopsis: How does Quarkus make your native executable smaller using UPX
+author: cescoffier
+---
+
+UPX is is an open-source, portable, high-performance *executable packer* initially created in 1996. 
+It takes an executable as input and produces a compressed executable. 
+Readers from a _certain_ age may have already used UPX a long time ago when programs needed to be compressed to fit on a floppy disk. 
+More than 20 years later, UPX is still rocking.
+
+In Quarkus 2.6, we integrated UPX compression in the Quarkus build. 
+So, Quarkus can automatically produce a compressed executable. 
+This post explains how you can use this new feature. 
+But, before going further, you need to understand that **nothing comes for free**. 
+The compressed executable is smaller on disk, but its memory usage is higher. 
+So make sure to read the "big warning" section before using it. 
+
+== Getting you ready
+
+Quarkus can compress your executable if you have installed UPX on your system or use an in-container build.
+
+UPX is available on most operating systems. 
+So, you should be able to download it from https://github.com/upx/upx/releases/tag/v3.96[their release page]. 
+UPX is cross-platform. 
+So you can compress a Linux 64 bits executable even from a Mac or Windows machine. 
```

Comment:
```suggestion
So you can compress a Linux 64 bits executable even from a macOS or Windows machine. 
```
I wouldn't have mentioned this one but since I added other comments :).